src/main/scala/it/pagopa/interop/agreementprocess/service/CatalogManagementService.scala

```diff
@@ -37,13 +37,24 @@ object CatalogManagementService {
     descriptorId: UUID,
     allowedStates: List[EServiceDescriptorState]
   ): Future[Unit] = {
-    val descriptorStatus = eService.descriptors.find(_.id == descriptorId).map(_.state)
+    val validDescriptorState = List(EServiceDescriptorState.SUSPENDED, EServiceDescriptorState.PUBLISHED)
```

Comment:
> shouldn't this actually be `allowedStates`?

`allowedStates`  in `validateActivationOnDescriptor` take also `ServiceDescriptorState.DEPRECATED`
is this intended?
the request is only for `EServiceDescriptorState.SUSPENDED` and `EServiceDescriptorState.PUBLISHED`

Reply:
In the `validateActivationOnDescriptor` function, `allowedStates` is also `EServiceDescriptorState.DEPRECATED`
is that okay?

Reply:
it's a fair point.
`DEPRECATED` is allowed because right now the Producer has the ability to approve (activate) and agreement created on a previous descriptor.

Thinking about it, the activation of a suspended agreement should probably not be enforced only on the latest descriptor.
For example:

1. EService E1 has only `ACTIVE` descriptor D1
2. Agreement A1 is created and activated for D1
3. D2 is created, D1 is deprecated
4. D1 is `SUSPENDED`
5. A1 is suspended
6. A1 is re-activated

This should be possible, even if D1 is not the latest version.
Given this point, it's possible that the activation should not follow the requirement of the latest EService Descriptor.

@beetlecrunch @TonioGela what do you think?

Reply:
Seems reasonable to me, as D2 and D1 may be profoundly different. Think about D1 being related to the v1 of an API interface while D2 being related to the v2 APIs. A consumer that suspended an agreement and that wants to reactivated will desire to reactivate an agreement over the v1 APIs that he was compatible with. Also the consumer might not have the attributes to subscribe to D2.

Reply:
can it work to insert an additional filter?

```
val latestDescriptor = eService.descriptors
                                       .find(_.id == descriptorId)
                                       .filterNot(_.state == EServiceDescriptorState.DRAFT)
                                       .filter(g => g.state == EServiceDescriptorState.PUBLISHED 
                                                    || g.state == EServiceDescriptorState.SUSPENDED)
                                       .maxByOption(_.version.toLong)
```

I thought it been deprecated or not, we must always fetch the latest version of the descriptor in published or suspended state.
So you can drop the deprecated states and change the error from `NotLatestEServiceDescriptor` to `DescriptorNotFound`

what do you think? @galales @TonioGela